.github/workflows/bump_dependencies.yml

```diff
@@ -0,0 +1,53 @@
+name: Bump Versions
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  schedule:
+    - cron: '30 17 * * WED' # 12:30 EST on wednesdays
+  workflow_dispatch:
+
+jobs:
+  bump-versions:
+    name: Bump Package Versions
+    if: github.repository_owner == 'viamrobotics'
+    runs-on: [self-hosted, x64]
+    container:
+      image: ghcr.io/viamrobotics/canon:amd64
+    timeout-minutes: 10
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.head_ref }}
+        token: ${{ secrets.GIT_ACCESS_TOKEN }}
+
+    - name: Update go dependencies
+      run: |
+        git init
+        git add .
+        sudo chown -R testbot .
+        sudo -u testbot bash -lc 'go get -u .'
+        GEN_DIFF=$(git status -s)
+
+        if [ -n "$GEN_DIFF" ]; then
+            echo 
+            git status
+            
+        else
+            echo "No dependency updates needed"
+            exit 1
```

Comment:
This is going to fail a run. You shouldn't fail just because no update happens. You want to just NOT do the next part. You should output a value in the first part of this if statement, (no need for the "else") then use an "if" in the following step to only do the PR if the value was set.

So something like `echo needs_pr=1 >> $GITHUB_OUTPUT` here, and then in the next step, `if: steps.goupdate.output.needs_pr == 1` (My syntax may not be perfect here.) Also, you'll need to add an `id: goupdate` field to this step (or whatever you want to id the step as.)

https://rmannibucau.metawerx.net/github-action-output-communicate-between-steps.html has a decent example.